_projects/2024/100508856/100508856.Rmd

```diff
@@ -0,0 +1,333 @@
+colnames(data_17) <- c("sovereignt","Hours")
+world_17 <- left_join(World,data_17,by="sovereignt")
+tm_shape(world_17) +
+  tm_polygons("Hours") +
+  tm_layout(
+    title ="Annual working hours per worker,2017",
+    title.size=0.7,
+    title.color="#008B8B",
+    title.position = c(0.25,0.97),
+    scale=1.5,
+    earth.boundary=T,
+    asp=1.75,
+    legend.show=T,
+    legend.only=F,
+    legend.outside=F,
+    legend.position=c(0.05,0.2),
+    legend.stack="horizontal",
+    legend.width=0.6,
+    legend.height=0.7,
+    legend.title.color="#8B1C62",
+    legend.title.size=0.7,
+    legend.text.color="#008B8B",
+    legend.text.size=0.5,
+    legend.frame=F
+  )
+
+```
+
+
+## Part two
+
+## evolution path
+
+The original graph is the evolutionary path of the data of different countries superimposed, and then dynamic simulation is carried out. In this part, I select several regions to draw one of them. Here is a link to the original graph:(https://ourworldindata.org/grapher/annual-working-hours-vs-gdp-per-capita-pwt?time=earliest..2019&country=BRA~USA~DEU~TWN%C2%AEion~PRT~ESP~AUS~NZL~ZAF)
+
+
+```{r}
+
+data <- read.csv( "C:/Users/liaom/Desktop/Courses/数据可视化/pre/123/GDP.csv")
+
+colnames(data)[c(4,5)] <- c("work_time","GDP")
+data1 <- data[-which(is.na(data$GDP)),]
+data2 <- data1[-which(is.na(data1$work_time)),]
+data3 <- dplyr::filter(data2, grepl('Brazil|China|France|Germany|Hong Kong|India|
+                                   Singapore|Taiwan|United Kingdom|United States', Entity))
+
+ggplot(data3,aes(GDP,work_time,color=Entity))+
+  geom_point(alpha=0.5)+
+  geom_line(alpha=0.5) +
+  scale_color_okabe_ito(name = "Country/Region")+
+  
+  labs(x = 'GDP per capita', y = 'Annual working hours per worker',color="Country/Region") +
+  scale_x_continuous(breaks = c(0,10000,20000,30000,40000,50000,60000,70000))+
+  theme_bw()+
+  theme(axis.text=element_text(size = 7),
+        axis.title.x =element_text(size = 10),
+        axis.title.y =element_text(size = 12))
+
+```
+
+
+## Alternative Representations
+
+## Data comparison chart
+
+In this section, new data has been added, specifically the GDP and population numbers of different countries. Consequently, the next image I replicated is a bubble chart(Figure 7), used to compare the relationship between the average annual working hours, the population size, and the GDP of various countries. Due to the large number of countries, this chart primarily displays a selection of countries and regions, represented in different colors.
+
+## Steps
+
+1. Read file data
+2. Extract and modify column names
+3. Clean data: Remove blank data
+4. Extract the country I want to draw
+5. Use the log10 function to draw the bubble, and use the ggplot2 function to set the x axis, y axis, color, title
+
+
+```{r}
+data <- read.csv("C:/Users/liaom/Desktop/Courses/数据可视化/pre/123/annual-working-hours-vs-gdp-per-capita-pwt/annual-working-hours-vs-gdp-per-capita-pwt.csv")
```

Comment:
This is different than before, right? Well... all CSVs should be added to the pull request. ;-)